src/fakes/ray/pubsub/subscriber.h

```diff
@@ -34,37 +34,22 @@ class FakeSubscriber : public SubscriberInterface {
  public:
   bool Subscribe(
       std::unique_ptr<rpc::SubMessage> sub_message,
-      const rpc::ChannelType channel_type,
+      rpc::ChannelType channel_type,
       const rpc::Address &owner_address,
-      const std::string &key_id,
+      const std::optional<std::string> &key_id,
```

Comment:
I'm personally not much of a fan of optionals being used like this...  Could we maybe use a pointer?  Then just do if(key_id) {...}?  That or overload the function...

I'll admit, maybe my bias comes from the travesty that is the java optional implementation, but I'm not sure I've seen a compelling reason for optionals outside of method return types.

Reply:
talked in person.  Oh fine, have it your way!